### PR TITLE
Pipeline v3 - Update Contributing Guide + Fix NPM Publishing

### PR DESCRIPTION
**Pull request by taymoork2**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -283,6 +287,27 @@ jobs:
...
+          command: |
+            # Grab the 2nd to last commit message from the log
+            # The last commit message should be [skip ci] from the version_bump job
+            LAST_MESSAGE=$(git log -n 1 --skip 1 --pretty=format:%s)
+            SECOND_LAST_MESSAGE=$(git log -n 1 --skip 1 --pretty=format:%s)
+
+            echo "$LAST_MESSAGE"
+            echo "$SECOND_LAST_MESSAGE"
+
+            if [[ "$LAST_MESSAGE" == *\["skip npm"\]* || "$SECOND_LAST_MESSAGE" == *\["skip npm"\]* ]]; then
```
Comment (adamweeks) [inline .circleci/config.yml]: Can we functionize this and use it in both checks?

File: .circleci/config.yml [hunk 2]
```diff
@@ -240,18 +240,30 @@ jobs:
     steps:
       - checkout_and_fetch_tags
       - restore_node_modules
+      - add_ssh_keys:
```
Comment (adamweeks) [inline .circleci/config.yml]: This can't be done in circle config?

**Comment by adamweeks** [comment]
Please split the the current version script changes into another commit.

File: .circleci/config.yml (hunk 2 above)
Comment (taymoork2) [reply .circleci/config.yml]: Nope it needs to be added manually (tested without it yesterday and it defaulted to a non-writable key)

File: .circleci/config.yml [hunk 3]
```diff
@@ -240,18 +240,33 @@ jobs:
...
+      - run:
+          name: 'Build Docs'
+          command: npm run build:docs
+      - run:
+          name: 'Publish Docs'
+          command: |
+            # Check if there's anything to commit before committing
+            if [ -n "$(git status --porcelain)" ]; then
+              git add docs/
+              git commit -m "docs(): update docs"
```
Comment (adamweeks) [inline .circleci/config.yml]: Can we move this after the version bump so our commit message would look something like 
```suggestion
              git commit -m "docs(API): update docs for v${VERSION}"
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -283,6 +295,9 @@ jobs:
       - run:
           name: 'Push tagged release to upstream/master'
           command: git push upstream v${VERSION}:v${VERSION}
+      - run:
+          name: 'Build Packages (again)'
```
Comment (adamweeks) [inline .circleci/config.yml]:
```suggestion
          name: 'Rebuild Packages (post-push)'
```

File: .circleci/config.yml (hunk 3 above)
Comment (taymoork2) [reply .circleci/config.yml]: I don't want to put a version on it, if in the case that a dev supplies a `[skip npm]` commit

Comment (taymoork2) [reply .circleci/config.yml]: I don't want to put a version on it, in case a dev supplies a `[skip npm]` commit

**Pull request merged**